pkg/operator/ceph/disruption/clusterdisruption/osd.go

```diff
@@ -77,53 +94,106 @@ func (r *ReconcileClusterDisruption) createDefaultPDBforOSD(namespace string) er
 		return errors.Errorf("failed to find the namespace %q in the clustermap", namespace)
 	}
 	pdbRequest := types.NamespacedName{Name: osdPDBAppName, Namespace: namespace}
-	pdb := &policyv1beta1.PodDisruptionBudget{
-		ObjectMeta: metav1.ObjectMeta{
-			Name:      osdPDBAppName,
-			Namespace: namespace,
-		},
-		Spec: policyv1beta1.PodDisruptionBudgetSpec{
-			MaxUnavailable: &intstr.IntOrString{IntVal: 1},
-			Selector: &metav1.LabelSelector{
-				MatchLabels: map[string]string{k8sutil.AppAttr: osdPDBAppName},
-			},
-		},
+	objectMeta := metav1.ObjectMeta{
+		Name:      osdPDBAppName,
+		Namespace: namespace,
 	}
-	ownerInfo := k8sutil.NewOwnerInfo(cephCluster, r.scheme)
-	err := ownerInfo.SetControllerReference(pdb)
-	if err != nil {
-		return errors.Wrapf(err, "failed to set owner reference to pdb %q", pdb)
+	selector := &metav1.LabelSelector{
+		MatchLabels: map[string]string{k8sutil.AppAttr: osdPDBAppName},
 	}
-
-	err = r.client.Get(context.TODO(), pdbRequest, &policyv1beta1.PodDisruptionBudget{})
+	usePDBV1, err := usePDBV1Version(r)
 	if err != nil {
-		if apierrors.IsNotFound(err) {
-			logger.Info("all PGs are active+clean. Restoring default OSD pdb settings")
-			logger.Infof("creating the default pdb %q with maxUnavailable=1 for all osd", osdPDBAppName)
-			return r.createPDB(pdb)
+		return err
+	}
+	if usePDBV1 {
```

Comment:
A suggestion on the pattern... Instead of a big if/else code block and a very small implementation for the func `createPDBV1()`, how about instead move all the version-specific code into the func such as `createPDBV1()`? For example, this method becomes simple:
```go
if usePDBV1 {
  return createPDBV1()
}
return createPDBV1Beta1()
```

Then later when we stop supporting K8s versions with v1beta1 we can more easily delete the dead code. This pattern seems good in other places for create or deleting the pdbs

Reply:
Yes, it was a good Pattern and is easily maintainable, but  `version-specific` code is different in different methods which lead to more duplication but.
Like: [pdb](https://github.com/rook/rook/blob/889a08273476acd5a1b06860f2385e3938291129/pkg/operator/ceph/disruption/clusterdisruption/osd.go#L86) vs [pdb](https://github.com/rook/rook/blob/889a08273476acd5a1b06860f2385e3938291129/pkg/operator/ceph/disruption/clusterdisruption/osd.go#L86)